### PR TITLE
Order api   get- wuth paging

### PR DESCRIPTION
**Pull request by ShulamitTol**
I hope I understood the meaning of the user id- its saved in every order document as employee?

### CODE REVIEW

File: src/main/java/com/sapred/ordermanagerred/controller/OrderController.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    @Autowired
+    private OrderService orderService;
+    //the params should be pathparams?...
+    @GetMapping("/allOrders/{userId}/{status}/{pageNumber}")
+    public List<Orders> getOrders(@PathVariable("userId") String userId, @PathVariable("status") String statusId, @PathVariable("pageNumber") int pageNumber) {
+        String companyId="11";//should be by the taken!
+        return orderService.getOrders(companyId,statusId,pageNumber,userId);
+    }
+    // זה פונקציה שמכניסה נתונים רק בשביל לבדוק שההתחברות עובדת
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/controller/OrderController.java:25]: comments only in english

File: src/main/java/com/sapred/ordermanagerred/model/Orders.java [hunk 2]
```diff
@@ -29,6 +29,15 @@ public class Orders {
     private Date ExpiryOn;
     private int cvc;
     private boolean notificationFlag;
-    @DBRef
     private  AuditData auditData;
+
+    public Orders(String s, String employee, String customer, int i,Company companyId, AuditData d,String orderStatusId) {
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/model/Orders.java:34]: you can add construct with Lombok

File: src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface OrderRepository extends MongoRepository<Orders, String> {
+    //also find by user id- employee field in the orders??????????
+    Page<Orders> findByCompanyId_IdAndOrderStatusIdAndEmployee(String companyId,String  orderStatusId,String employee, Pageable pageable);
+
+//    Page<Orders> findByCompanyId_IdAndOrderStatusId(String companyId,String  orderStatusId, Pageable pageable);
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java:14]: please remove commented lines

File: src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import com.sapred.ordermanagerred.model.Orders;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface OrderRepository extends MongoRepository<Orders, String> {
+    //also find by user id- employee field in the orders??????????
+    Page<Orders> findByCompanyId_IdAndOrderStatusIdAndEmployee(String companyId,String  orderStatusId,String employee, Pageable pageable);
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java:12]: Did you check it?
I think under score is not correct

File: src/main/java/com/sapred/ordermanagerred/controller/OrderController.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.List;
+
+@RestController
+@RequestMapping("/Order")
+public class OrderController
+{
+    @Autowired
+    private OrderService orderService;
+    //the params should be pathparams?...
+    @GetMapping("/allOrders/{userId}/{status}/{pageNumber}")
```
Comment (moriyahazan) [inline src/main/java/com/sapred/ordermanagerred/controller/OrderController.java:20]: change from allOrders to getOrders

**Review by moriyahazan — APPROVED** [review]
Great Job
please review my comments and fix

**Comment by moriyahazan** [comment]
Please add also exception handling in controller

File: src/main/java/com/sapred/ordermanagerred/model/Orders.java (hunk 2 above)
Comment (ShulamitTol) [reply src/main/java/com/sapred/ordermanagerred/model/Orders.java:34]: it a ctor with just part of the args, can I also add it with lombok? how?

File: src/main/java/com/sapred/ordermanagerred/controller/OrderController.java (hunk 1 above)
Comment (ShulamitTol) [reply src/main/java/com/sapred/ordermanagerred/controller/OrderController.java:25]: got it!

File: src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java (hunk 3 above)
Comment (ShulamitTol) [reply src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java:14]: ok!

File: src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java (hunk 4 above)
Comment (ShulamitTol) [reply src/main/java/com/sapred/ordermanagerred/repository/OrderRepository.java:12]: yes,it works. I found this way to  access a nested feild.

File: src/main/java/com/sapred/ordermanagerred/controller/OrderController.java (hunk 5 above)
Comment (ShulamitTol) [reply src/main/java/com/sapred/ordermanagerred/controller/OrderController.java:20]: I am going to change it

**Comment by ShulamitTol** [comment]
hi,
thanks!
I got your comments - I wrote responses there.
so after the fixes I merge it to the main. ok?
have a good day

‫בתאריך יום א׳, 2 ביולי 2023 ב-10:09 מאת ‪Moriya‬‏ <‪
***@***.***‬‏>:‬

File: src/main/java/com/sapred/ordermanagerred/controller/OrderController.java (hunk 5 above)
Comment (moriyahazan) [reply src/main/java/com/sapred/ordermanagerred/controller/OrderController.java:20]: good

**Review by moriyahazan — CHANGES_REQUESTED** [review]

**Pull request merged**